docs/general/dql/selects.rst

```diff
@@ -841,6 +841,50 @@ Object property can also be addressed in the :ref:`where clause
     +------------+----------------------------------------------------------------------+
     SELECT 1 row in set (... sec)
 
+For selecting records having values with empty or unassigned objects, like::
+
+    cr> insert into locations (id, name, position, kind, inhabitants) values
+    ... (99, 'Voidspace', 0, 'Milliways', {});
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty objects, is::
+
+    cr> select count(*) from locations where inhabitants is null;
+    +----------+
+    | count(*) |
+    +----------+
+    |       12 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+.. Hidden: drop record again
+
+    cr> delete from locations where id=99;
+    DELETE OK, 1 row affected  (... sec)
+
+For determining the number of items in an object, you can use the
+:ref:`object_keys <scalar-object_keys>` and :ref:`array_length
+<scalar-array_length>` scalar functions::
```

Comment:
I think there should be some elaboration why `IS NULL` matches on empty objects. I'm also not sure if this is generally the case (e.g. sys tables, or if the column doesn't have an index)

Reply:
Judging from your recent adjustment #12704, I am assuming the same ambiguity might get fixed on this end as well. So I will adjust the patch to use a different expression instead.
```sql
WHERE inhabitants IS NULL OR inhabitants = {}
```